src/XRP/default-xrp-client.ts

```diff
@@ -456,6 +458,43 @@ export default class DefaultXrpClient implements XrpClientDecorator {
     return await this.getTransactionResult(transactionHash)
   }
 
+  /**
+   * Enables DepositPreauth and authorizes an XRPL account to deposit to this XRPL account.
+   *
+   * @set https://xrpl.org/depositpreauth.html
+   *
+   * @param xAddressToAuthorize The X-Address of the account to enable DepositPreauth for.
+   * @param wallet The wallet associated with the XRPL account enabling DepositPreauth and that will sign the request.
+   */
+  public async authorizeDepositPreauth(
```

Comment:
this name is weird, but since the docs specify that you can only either authorize or unauthorize, it felt useful to have separate apis for the two instead of a more abstract function that does both so you can't hurt yourself in that way. let me know what you think. feels like it could also be `depositPreauthAuthorize`

Reply:
> it felt useful to have separate apis for the two instead of a more abstract function that does both so you can't hurt yourself in that way.

Agree.

Reply:
I also agree.  I would perhaps name them `authorizeSendingAccount` or `authorizePayer` / `unauthorizeSendingAccount` etc.   There's no reason we have to stick to "DepositPreauth" schema.

Reply:
good point @amiecorso - i'll try both of those out and see how it feels